syntax = "proto3";

package erabikata;

enum AnalyzerMode {
  SudachiA = 0;
  SudachiB = 1;
  SudachiC = 2;
}

message AnalyzeRequest {
  string text = 1;
  AnalyzerMode mode = 2;
}

message AnalyzedResponse {
  repeated Word words = 1;

  message Word {
    string baseForm = 1;
    string dictionaryForm = 2;
    string reading = 3;
    string original = 4;
  }
}

service AnalyzerService {
  rpc AnalyzeText(AnalyzeRequest) returns (AnalyzedResponse) {}

  rpc AnalyzeBulk(stream AnalyzeRequest) returns (stream AnalyzedResponse) {}
}